concepts/string.md

```diff
@@ -0,0 +1,36 @@
+---
+description: String
+---
+
+# String
+
+A string is an object in memory that stores a sequence of characters and colloquially it is the representation of text in the computer.
+Examples:
+1. `bicycle`
+2. `A man with a telescope 🔭 from the city of Třebíč`
+3. ```
+   Introduction: Lorem Ipsum is simply dummy text of the printing and typesetting industry.
+   
+   Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. 
+   ```
+
+The first example is simply a 7-letter word of English letters.
+The second example shows that a string can contain spaces, letters from non-English alphabets and even emojis.
+The last example demonstrates that even paragraphs with special whitespace characters, such as the newline, usually written as `\n` are stored as a string.
+
+## Encoding
+
+A particular focus is paid to how the characters are encoded in the string.
```

Comment:
I'd keep this all simpler and less detailed, and order the flow differently.

> In software, every character has a unique identifier.
> 
> > [ASCII](https://en.wikipedia.org/wiki/ASCII) codes
> > | Character      | Decimal representation |
> > | ----------- | ----------- |
> > | ...      | ...       |
> > | `A`      | 65       |
> > | `B`      | 66        |
> > | `C`      | 67        |
> > |  ...     |  ...        |
>
> ASCII, an early encoding standard, only has 128 characters.  It includes the digits, the letters of the Latin alphabet, common punctuation and control characters.  It does not include `β`, `喂`, or even `é`.
>
> { describe the problems with multiple languages and conflicting standards here }
>
>
> **UTF-8** ...
>
> Other types of encoding:
> - **HTML encoding**
> - **JSON encoding**
> 
> ## Escaping
> ...

Reply:
> Other types of encoding:
>    HTML encoding
>    JSON encoding
This is something different. HTML & JSON encoding are on top of character representation which is handled by the system/standard library.

I am honestly not sure how describe escaping intuitively and well. Maybe you have some suggestions?

Nevertheless I restructured this part.

Reply:
Definitely HTML encoding etc are different, but in practice since it's less low-level, people encounter it more in their work.

Probably we need to make a separate article on encoding, and another on escaping.

I don't want to delay your PR with that.

Reply:
I meant to replace

> A particular focus is paid to how the characters are encoded in the string.

with

> In software, every character has a unique identifier.

Let's keep in mind that this article is for people who don't know what a string is!

(Here, you're using "encoded" to explain what encoding is.)

And then the point of the ASCII table is just to give an example of giving a unique id to every character.